pkg/db/db_client/db_client_connect.go

```diff
@@ -74,5 +80,29 @@ func (c *DbClient) establishConnectionPool(ctx context.Context) error {
 		return err
 	}
 	c.pool = dbPool
+
+	return c.establishSystemConnectionPool(ctx, config)
+}
+
+// establishSystemConnectionPool creates a connection pool that can be used to support execution of user-initiated
```

Comment:
// establishSystemConnectionPool creates a connection pool to use to execute 
// system-initiated queries (loading of connection state etc.)
// unlike establishConnectionPool, which is run first to create the user-query pool
// this doesn't wait for the pool to completely start, as establishConnectionPool will have established and verified a connection with the service
func (c *DbClient) establishSystemConnectionPool(ctx context.Context, config *pgxpool.Config) error {